extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js

```diff
@@ -106,55 +114,190 @@ export class AmpAdNetworkFakeImpl extends AmpA4A {
       root.setAttribute('amp4ads', '');
     }
 
-    // Remove all AMP scripts.
+    const creative = root./*OK*/ outerHTML;
+    const creativeSplit = creative.split('</body>');
+    const docWithMetadata =
+      creativeSplit[0] +
+      `<script type="application/json" amp-ad-metadata>` +
+      metadata +
+      '</script></body>' +
+      creativeSplit[1];
+    return docWithMetadata;
+  }
+
+  /**
+   * Generates metadata for AMP4ADS doc
+   * @param {Document} doc
+   * @return {string}
+   */
+  generateMetadata_(doc) {
```

Comment:
please break up the method to smaller functions for readability
also, think about moving this code to a separate file in case we want to port this code to server side.